### PR TITLE
updating billing to include gateway fees

### PR DESCRIPTION
**Pull request by Rohit25negi**
http://manage.dimagi.com/default.asp?280243
https://docs.google.com/document/d/16aH39sj7OixeYI_FwiJH3lmJCCMI4ZJzhn-kZ8bucUs/edit?ts=5b47a681#heading=h.o4xghqm2btkm

### CODE REVIEW

**Comment by Rohit25negi** [comment]
@nickpell @calellowitz  please review the changes. Are these enough for billing change for UCLA project?. 

PS: unittest cases under writing.

File: corehq/apps/smsbillables/tests/test_gateway_fees.py [hunk 1]
```diff
@@ -233,6 +233,24 @@ def test_no_matching_fee(self):
...
+        arbitrary_message = messages[randint(0, len(messages))]
+
+        toggles.ENABLE_INCLUDE_SMS_GATEWAY_CHARGING.set(
+            arbitrary_message.domain, True, toggles.NAMESPACE_DOMAIN)
+        billable = SmsBillable.create(arbitrary_message)
+        toggles.ENABLE_INCLUDE_SMS_GATEWAY_CHARGING.set(
+            arbitrary_message.domain, False, toggles.NAMESPACE_DOMAIN)
+
+        self.assertIsNotNone(billable)
+        self.assertIsNotNone(billable.gateway_fee)
```
Comment (nickpell) [inline corehq/apps/smsbillables/tests/test_gateway_fees.py]: I would also be good to check that the gateway fee charge assigned is correct.

Here's an example of how to mock a response from the twilio api:

https://github.com/dimagi/commcare-hq/blob/1bfe88ffe51eee2424a48e998b5f09926096f2c3/corehq/apps/smsbillables/tests/test_gateway_fees.py#L236-L272

**Review by nickpell — CHANGES_REQUESTED** [review]
Implementation of the feature looks good.  Had one request regarding tests.

**Review by nickpell — APPROVED** [review]

**Comment by nickpell** [comment]
@Rohit25negi I just got a travis test failure:

```
ERROR: corehq.apps.smsbillables.tests.test_gateway_fees:TestGatewayFee.test_fee_for_non_global_backend

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/mnt/commcare-hq-ro/corehq/apps/smsbillables/tests/test_gateway_fees.py", line 242, in test_fee_for_non_global_backend

    arbitrary_message = messages[randint(0, len(messages))]

IndexError: list index out of range
```

I think this will happen every 1 out of ```len(messages)``` times tests are run.  Can you please fix?

**Comment by Rohit25negi** [comment]
@nickpell can you share the travis job url

**Comment by Rohit25negi** [comment]
Oh got it. @nickpell  Thanks for pointing it out.

**Pull request merged**